src/index.ts

```diff
@@ -224,13 +224,17 @@ function verbify(verb) {
   };
 }
 
-function scheduleProgressEvent(request, startTime, totalTime) {
+// Creates fake progress events every 50ms until the request is complete
+function scheduleProgressEvent(request) {
   setTimeout(function() {
     if (!request.aborted && !request.status) {
-      var ellapsedTime = new Date().getTime() - startTime.getTime();
-      request.upload._progress(true, ellapsedTime, totalTime);
-      request._progress(true, ellapsedTime, totalTime);
-      scheduleProgressEvent(request, startTime, totalTime);
+      if (request && request.requestBody && request.requestBody.size) {
+        var size = request.requestBody.size;
+        request.upload._progress(true, size, size);
```

Comment:
Is this going to always have progress be 100%?

Reply:
I think the issue here is after `totalTime`, the progress is still being scheduled. Maybe the fix should be stopping scheduling after `totalTime`? At that time the request has been resolved https://github.com/pretenderjs/pretender/blob/master/src/index.ts#L354.

Reply:
Yes, this would always have progress be at 100. I thought about having a progression based on time but decided that faking progress events at 100% would be just as good (certainly better than time diffs) for most use-cases.

As for the scheduling, you are correct that `scheduleProgressEvent` will fire after the `totalTime` (after response sent) but it stops itself because the request will then have a status: https://github.com/pretenderjs/pretender/blob/master/src/index.ts#L229

The current way it's stopping itself is fine and wasn't a problem for my use case.